optuna/trial/_trial.py

```diff
@@ -638,29 +626,30 @@ def set_system_attr(self, key, value):
     def _suggest(self, name, distribution):
         # type: (str, BaseDistribution) -> Any
 
-        if self._is_fixed_param(name, distribution):
-            param_value = self.storage.get_trial_system_attrs(self._trial_id)["fixed_params"][name]
-        elif self._is_relative_param(name, distribution):
-            param_value = self.relative_params[name]
-        else:
-            trial = self.storage.get_trial(self._trial_id)
-
-            study = pruners._filter_study(self.study, trial)
+        storage = self.storage
+        trial_id = self._trial_id
 
-            param_value = self.study.sampler.sample_independent(study, trial, name, distribution)
+        trial = storage.get_trial(trial_id)
 
-        return self._set_new_param_or_get_existing(name, param_value, distribution)
-
-    def _set_new_param_or_get_existing(self, name, param_value, distribution):
-        # type: (str, Any, BaseDistribution) -> Any
+        if name in trial.distributions:
+            # No need to sample if already suggested.
+            distributions.check_distribution_compatibility(trial.distributions[name], distribution)
+            param_value = distribution.to_external_repr(storage.get_trial_param(trial_id, name))
+        else:
+            if self._is_fixed_param(name, distribution):
+                param_value = storage.get_trial_system_attrs(trial_id)["fixed_params"][name]
+            elif distribution.single():
+                param_value = distributions._get_single_value(distribution)
```

Comment:
This is optional, but how about adding a test case to raise warnings if single-value distributions are given with fixed params that are out of ranges? This code correctly handles such a case, and I propose it for future development.

Reply:
Thank you for the suggestion. I added tests for `enqueue_trial` since it's relevant (although a bit stretched) to this PR.